src/pybind/mgr/prometheus/module.py

```diff
@@ -536,7 +554,8 @@ def collect(self) -> None:
                     sleep_time = 0
 
                 with self.mod.collect_lock:
-                    self.mod.collect_cache = data
+                    self.mod.collect_cache = gzip.compress(data.encode("utf-8"))
```

Comment:
If I'm not wrong, what we're doing here is compressing this data to save memory and serve it faster through HTTP, right? In that case, I think that we should add a `Content-Encoding: gzip` in the HTTP header.

BTW, @rkachach (FYI) was recently doing a fix, since `cherrypy` already has built-in support for `gzip`.

Reply:
Nice, I took some ideas from that PR and implemented them in those changes